### PR TITLE
Added CSV mixin to wrap `ripe-commons-js`

### PR DESCRIPTION
**Pull request by veryprofessionaldodo**
| - | - |
| --- | --- |
| Issue | Related to [#302](https://github.com/ripe-tech/ripe-pulse/issues/302) |
| Dependencies | https://github.com/ripe-tech/ripe-commons-js/pull/33, https://github.com/ripe-tech/ripe-commons-js/pull/35 |
| Decisions | Some functions were extracted from `ripe-util-vue`, but a mixing would be useful to wrap them neatly. This PR adds a new mixin to handle that. |

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-*
+* Add CSV mixin to wrap `ripe-commons-js` functions - related to [#302](https://github.com/ripe-tech/ripe-pulse/issues/302)
```
Comment (joao-conde) [inline CHANGELOG.md:12]: ripe-pulse/#302

File: vue/mixins/csv.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { objectToCsv, readCsv, buildCsv, parseCsv, parseCsvComplex } from "ripe-commons";
```
Comment (joao-conde) [inline vue/mixins/csv.js:1]: Alphabetic order

File: vue/test/mixins/csv.test.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
+const assert = require("assert");
+const mixins = require("../../mixins");
+
+describe("csvMixin", () => {
```
Comment (joao-conde) [inline vue/test/mixins/csv.test.js:4]: Move these tests to RIPE Commons, as this is only a wrapper around it

**Review by joao-conde — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (veryprofessionaldodo) [reply CHANGELOG.md:12]: Welp, that's clearly wrong :) 

Nice catch, thanks! It's another #302 clearly :D

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-*
+* Add CSV mixin to wrap `ripe-commons-js` functions - related to [ripe-pulse/#302](https://github.com/ripe-tech/ripe-pulse/issues/302)
```
Comment (joao-conde) [inline CHANGELOG.md:12]: No need for the "related to"

**Review by joao-conde — CHANGES_REQUESTED** [review]

**Review by joao-conde — APPROVED** [review]

**Review by BeeMargarida — APPROVED** [review]

File: vue/mixins/csv.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+        buildCsv(data, headers = []) {
+            return buildCsv(data, headers);
+        },
+        parseCsv(dataS, object = false, sanitize = true, delimiter = ",") {
+            return parseCsv(dataS, object, sanitize, delimiter);
+        },
+        parseCsvComplex(dataS, object = false, sanitize = true, delimiter = ",") {
+            return parseCsvComplex(dataS, object, sanitize, delimiter);
+        },
+        readCsv(file, parser = null) {
```
Comment (joamag) [inline vue/mixins/csv.js:14]: let's change this to `parseCsvFile` instead

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by joamag** [comment]
@veryprofessionaldodo let's use this refactor to change some of the function names :)

**Comment by joamag** [comment]
Let's move this mixin to `ripe-commons-vue` instead it makes more sense as this does not impact anything visual @veryprofessionaldodo  @joao-conde

**Comment by veryprofessionaldodo** [comment]
@joamag This means `ripe-commons-vue`will have to be added as a dependency to at least `ripe-util-vue`, that's not an issue, right?

**Comment by joao-conde** [comment]
@veryprofessionaldodo Makes sense to me that ripe-util-vue, being a vue project, uses our ripe-commons-vue, and likewise makes sense that ripe-commons-vue, being a javascript project, makes use of ripe-commons-js. In my opinion 👍

**Comment by joamag** [comment]
@veryprofessionaldodo Yes the dependency sequence makes sense

**Pull request closed without merging**